course/src/main/java/org/openedx/course/presentation/container/CourseContainerFragment.kt

```diff
@@ -205,21 +210,44 @@ class CourseContainerFragment : Fragment(R.layout.fragment_course_container) {
                                 }
 
                                 CalendarSyncDialogType.OUT_OF_SYNC_DIALOG -> {
+                                    viewModel.logCalendarSyncUpdate(true)
                                     viewModel.addOrUpdateEventsInCalendar(
                                         updatedEvent = true,
                                     )
                                 }
 
                                 CalendarSyncDialogType.EVENTS_DIALOG -> {
+                                    viewModel.logCalendarSyncedConfirmation(true)
                                     viewModel.openCalendarApp()
                                 }
 
                                 CalendarSyncDialogType.LOADING_DIALOG,
-                                CalendarSyncDialogType.NONE -> {
+                                CalendarSyncDialogType.NONE,
+                                -> {
                                 }
                             }
                         },
-                        dismissSyncDialog = {
+                        dismissSyncDialog = { dialog ->
+                            when (dialog) {
+                                CalendarSyncDialogType.SYNC_DIALOG ->
+                                    viewModel.logCalendarAddDates(false)
+
+                                CalendarSyncDialogType.UN_SYNC_DIALOG ->
+                                    viewModel.logCalendarRemoveDates(false)
+
+                                CalendarSyncDialogType.OUT_OF_SYNC_DIALOG ->
+                                    viewModel.logCalendarSyncUpdate(false)
+
+                                CalendarSyncDialogType.EVENTS_DIALOG ->
+                                    viewModel.logCalendarSyncedConfirmation(false)
+
+                                CalendarSyncDialogType.LOADING_DIALOG,
+                                CalendarSyncDialogType.PERMISSION_DIALOG,
+                                CalendarSyncDialogType.NONE,
+                                -> {
```

Comment:
nit: change {} to Unit

you can skip this

Reply:
replaced with `else -> {}`